pkg/infrastructure/openstack/clusterapi/clusterapi.go

```diff
@@ -89,6 +89,15 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 		return fmt.Errorf("failed to get OSPCluster: %w", err)
 	}
 
+	bootstrapMachine := &capo.OpenStackMachine{}
```

Comment:
The bootstrap node is not yet created when the `InfraReady` hook is called. You need to use `PostProvision` hook instead.